src/components/NavBar.jsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { useState } from 'react';
+import { useNavigate, useLocation } from 'react-router-dom';
+import BackcountryLogo from '../logos/backcountry-logo-with-text.svg';
+import NavBarSwitchingIcon from '../logos/NavBarSwitchingIcon.svg';
+import useSessionStorage from "../hooks/useSessionStorage";
+import Button from './Button';
+
+function NavBar() {
+    const navigate = useNavigate();
+    const location = useLocation();
+    const [showPopup, setShowpopup] = useState(false)
+    const [name] = useSessionStorage("userName");
+    const handlPopup=()=>{
+        setShowpopup(!showPopup)
+    }
+    
+    return (
+    <nav className="bg-white mt-2 mb-2">
+    <div className="flex flex-wrap justify-between items-center mx-auto max-w-screen-xl">
+    <div className="flex items-center ml-15">
+    <img 
+       className="ml-8 px-1 py-1 h-26 w-147"
+       src={BackcountryLogo} alt="Backcountry SVG"/>
+    </div>
+        <div className="flex items-center mr-10">
+           {location.pathname !== '/menuChooser' && (<div className='mr-2'>
```

Comment:
Why not merge everything from #27 to #38 in a single loop ?

Reply:
addressed